src/db_bench/db_bench.rs

```diff
@@ -0,0 +1,303 @@
+use bytes::Bytes;
+use leaky_bucket::RateLimiter;
+use rand::{RngCore, SeedableRng};
+use rand_xorshift::XorShiftRng;
+use slatedb::config::WriteOptions;
+use slatedb::db::Db;
+use std::collections::VecDeque;
+use std::sync::{Arc, Mutex};
+use std::time::Duration;
+use tokio::time::Instant;
+
+pub trait KeyGenerator: Send {
+    fn next_key(&mut self) -> Bytes;
+}
+
+// TODO: implement other distributions
+
+pub struct RandomKeyGenerator {
+    key_bytes: usize,
```

Comment:
Super nitty, but I didn't catch that this was a `usize` and was trying to figure out why you had a key_bytes byte array here. 😝  `key_len_bytes` would have helped.